engines/private/private.cpp

```diff
@@ -0,0 +1,1081 @@
+    if (_dossierNextSuspectMask == NULL)
+        return false;
+
+    if (inMask(_dossierNextSuspectMask->surf, mousePos)) {
+        if ((_dossierSuspect + 1) < _dossiers.size()) {
+            _dossierSuspect++;
+            _dossierPage = 0;
+            loadDossier();
+            drawMask(_dossierNextSuspectMask->surf);
+            drawMask(_dossierPrevSuspectMask->surf);
+        }
+        return true;
+    }
+    return false;
+}
+
+bool PrivateEngine::selectDossierPrevSuspect(Common::Point mousePos) {
+    if (_dossierPrevSuspectMask == NULL)
+        return false;
+
+    if (inMask(_dossierPrevSuspectMask->surf, mousePos)) {
+        if (_dossierSuspect > 0) {
+            _dossierSuspect--;
+            _dossierPage = 0;
+            loadDossier();
+            drawMask(_dossierNextSuspectMask->surf);
+            drawMask(_dossierPrevSuspectMask->surf);
+        }
+        return true;
+    }
+    return false;
+}
+
+void PrivateEngine::selectLoadGame(Common::Point mousePos) {
+    if (_loadGameMask == NULL)
+        return;
+
+    if (inMask(_loadGameMask->surf, mousePos)) {
+        loadGameDialog();
+    }
+}
+
+void PrivateEngine::selectSaveGame(Common::Point mousePos) {
+    if (_saveGameMask == NULL)
+        return;
+
+    if (inMask(_saveGameMask->surf, mousePos)) {
+        saveGameDialog();
+    }
+
+}
+
+bool PrivateEngine::hasFeature(EngineFeature f) const {
+    return (f == kSupportsReturnToLauncher);
+}
+
+void PrivateEngine::restartGame() {
+    debugC(1, kPrivateDebugFunction, "restartGame");
+
+    for (NameList::iterator it = variableList.begin(); it != variableList.end(); ++it) {
+        Private::Symbol *sym = variables.getVal(*it);
+        if (strcmp("kAlternateGame", sym->name->c_str()) != 0)
+            sym->u.val = 0;
+    }
+
+    // Diary
+    for (NameList::iterator it = locationList.begin(); it != locationList.end(); ++it) {
+        Private::Symbol *sym = locations.getVal(*it);
+        sym->u.val = 0;
+    }
+    inventory.clear();
+    _dossiers.clear();
+
+    // Sounds
+    _AMRadio.clear();
+    _policeRadio.clear();
+    _phone.clear();
+    _playedPhoneClips.clear();
+
+    // Movies
+    _repeatedMovieExit = NULL;
+    _playedMovies.clear();
+
+}
+
+Common::Error PrivateEngine::loadGameStream(Common::SeekableReadStream *stream) {
+    Common::Serializer s(stream, nullptr);
+    debugC(1, kPrivateDebugFunction, "loadGameStream");
+    _nextSetting = &kStartGame;
+    int val;
+
+    for (NameList::iterator it = variableList.begin(); it != variableList.end(); ++it) {
+        s.syncAsUint32LE(val);
+        Private::Symbol *sym = variables.getVal(*it);
+        sym->u.val = val;
+    }
+
+    // Diary
+    for (NameList::iterator it = locationList.begin(); it != locationList.end(); ++it) {
+        s.syncAsUint32LE(val);
+        Private::Symbol *sym = locations.getVal(*it);
+        sym->u.val = val;
+    }
+    uint32 size = 0;
+    size = stream->readUint32LE();
```

Comment:
you could write right away `uint32 size = stream->readUint32LE();`. Saves vertical space.

Reply:
Fixed.